modules/thunder_article/src/ThunderNodeForm.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Drupal\thunder_article;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\NodeForm;
+
+/**
+ * Base for handler for node edit forms.
+ */
+class ThunderNodeForm extends NodeForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $element = parent::actions($form, $form_state);
+
+    if ($element['publish']['#access'] && \Drupal::currentUser()->hasPermission('administer nodes')) {
+      $element['save_continue'] = $element['publish'];
```

Comment:
When you create new article and you click "Save and continue" -> Article will be automatically published. I'm not sure is that wanted behavior, I think article should stay unpublished by default. Also I'm not sure how good is this option during editing of already published article, because changes will be published.

Anyway, when article is edited, "Save and continue" should not change state of article (published, unpublished). When you click "Save and continue" on unpublished article it should stay unpublished, if you click same option on published article it should stay published.